packages/language/test/linking.test.ts

```diff
@@ -181,17 +243,119 @@ describe("Linking tests", () => {
 
  PUT(A.<|b>B);`));
 
-    /**
-     * TODO: Implement star handling in structured names
-     */
-    test.skip("Star name in structure should be qualifiable", () =>
+    test.skip("Star name in structure should result in partial qualification", () =>
       expectLinks(`
  DCL 1 A,
         2 *,
           3 B CHAR(8) VALUE("B"),
         2 <|b:B|> CHAR(8) VALUE("B2");
 
  PUT(A.<|b>B);`));
+
+    test("Must work before declaration", () =>
+      expectLinks(`
+ PUT(<|a>A);
+ <|a>A = "A2";
+ DCL <|a:A|> CHAR(8) INIT("A");
+ PUT(<|a>A);`));
+
+    test("Must work across procedure boundaries", () =>
+      expectLinks(`
+ PUT(<|a>A);
+
+ LABL: PROCEDURE;
+   PUT(<|a>A);
+ END LABL;
+
+ PUT(<|a>A);
+ CALL LABL;
+
+ DCL <|a:A|> CHAR(8) INIT("A");`));
+  });
+
+  /**
+   * These test probably should not in the linking tests, but I'm putting them here for reference for now.
+   */
+  describe("Faulty cases", () => {
+    test.skip("Redeclaration of label must fail", () => {
+      const doc = parseAndLink(`
+ OUTER: PROCEDURE;
+ END OUTER;
+ OUTER: PROCEDURE;
+ END OUTER;`);
+      assertDiagnostic(doc, {
+        code: PLICodes.Severe.IBM1916I.fullCode,
+        severity: Severity.S,
+      });
+    });
+
+    test("Unused label should warn", () => {
+      const doc = parseAndLink(`
+ OUTER: PROCEDURE;
+ END OUTER;`);
+      assertDiagnostic(doc, {
+        code: PLICodes.Warning.IBM1213I.fullCode,
+        severity: Severity.W,
+      });
+    });
+
+    /**
+     * TODO: Should this be tested here or in the validation step?
```

Comment:
Given the diagnostic is available at this stage, it's ok to check it here as well.